Proofread UPGRADE guide (#58899)
symfony/symfony: UPGRADE-7.2.md

If you're upgrading from a version below 7.1, follow the [7.1 upgrade guide](UPGRADE-7.1.md) first.

Table of Contents
-----------------

Bundles

 * [FrameworkBundle](#FrameworkBundle)

Bridges

 * [TwigBridge](#TwigBridge)

Components

 * [Cache](#Cache)
 * [Console](#Console)
 * [DependencyInjection](#DependencyInjection)
 * [Form](#Form)
 * [HttpFoundation](#HttpFoundation)
 * [Ldap](#Ldap)
 * [Lock](#Lock)
 * [Mailer](#Mailer)
 * [Notifier](#Notifier)
 * [Routing](#Routing)
 * [Security](#Security)
 * [Serializer](#Serializer)
 * [Translation](#Translation)
 * [Webhook](#Webhook)
 * [Yaml](#Yaml)

Cache
-----

>  * [BC BREAK] `igbinary_serialize()` is no longer used instead of `serialize()` when the igbinary extension is installed,

Reviewer: Why is it a BC break?

Author: Thanks for bringing this up!

I was reading https://github.com/symfony/symfony/pull/57702#issuecomment-2241598355 As far as I understood, people might have been installing igbinary for cache performance reasons. In 7.2, there is no way - as a framework user - to enable igbinary usage again. These applications are affected by a performance penalty, even if they were not at risk with the behavioral incompatibility.

Maybe BC break is too harsh of a classification, but this does sound more than just "oh something is deprecated" to me.

Reviewer: Maybe

```suggestion
 * `igbinary_serialize()` is no longer used instead of `serialize()` when the igbinary extension is installed (performance might be impacted),
```

DependencyInjection
-------------------

 * Deprecate `!tagged` Yaml tag, use `!tagged_iterator` instead

   *Before*
   ```yaml
   services:
       App\Handler:
           tags: ['app.handler']

       App\HandlerCollection:
           arguments: [!tagged app.handler]
   ```

   *After*
   ```yaml
   services:
       App\Handler:
           tags: ['app.handler']

>    App\HandlerCollection:
>        arguments: [!tagged_iterator app.handler]

Reviewer 2: Indentation seems inexact here
```suggestion
       App\HandlerCollection:
           arguments: [!tagged_iterator app.handler]
```
